src/com/kandbnutrition/model/UserData.java

```diff
@@ -0,0 +1,56 @@
+package com.kandbnutrition.model;
+
+
+/*
+ * Created 1/30/2016 by Kyle Wolff 
+ * 
+ * This class is a singleton class to store the users data. 
+ * 
+ * The uID is the main String here because that is what Identifies the user for different calls
+ * that will be made for the Fittracker. 
+ */
+
+public class UserData {
+	
+	private static UserData userData;
+	private String email, firstName, lastName, uID;
+	
+	protected UserData() {}
+	
+	public static synchronized UserData getUserData() {
+		
+		if(userData == null) {
+			userData = new UserData();
+		}
+		
+		return userData;
+	}
```

Comment:
```
public enum UserData
{
     INSTANCE;

     UserData() { } /*Constructor*/
}
```
IIRC:
That is how a enum is used for a Singleton (always recommended from what I read).  Reason why it is thread-safe is that sometimes two different threads will call ```getUserData()```.  You can have a case where both threads check the condition ```if(userData == null)``` at the same time and both pass.  Now Thread 1 gets to ```userData = new UserData()``` and succeeds.  Then, Thread 2 gets to the same line of code and creates a new UserData object.  This totally defeats the original goal of a Singleton because now we have two separate objects.